src/header.rs

```diff
@@ -585,8 +507,6 @@ mod tests {
     }
 
```

Comment:
My comments about negative/large integers above have made me realize that we are missing tests for negative integers and integers larger than u64.